### DISCUSSION TITLE
text-based alt editor

### DISCUSSION THREAD

**Discussion by raynepaws** [cytrinc/bark-coding#175] [New Features (not block requests)]
for users wanting a more advanced experience, there could be a text-based editor alternative to the block-based one. i would be willing to develop this, as i already have a basic skeleton for it.

i would make a new programming language for bark, and it would basically look like this:
```
start {
  new sprite square = rect 10 20;
  loop {
    square.move_x 10;
  }
}
```
this code would create a 10 by 20 pixel rectangular sprite and move it on the x axis 10 pixels every frame

this new code editor would include:
- code coloring
- extensive documentation
- easy-to-read programming language (name pending)
- a terminal (wow)

what do y'all think?? :)

**Comment by raynepaws** [comment]
i'm sorry about my inactivity recently. i'm gonna start working on this

**Reply by nmsderp** [reply]
just realized this isn't very hard, just allow editing the generated p5.js code

**Reply by raynepaws** [reply]
> just realized this isn't very hard, just allow editing the generated p5.js code

i'm not using p5.js
